src/cpptools.ts

```diff
@@ -119,37 +119,38 @@ function parseTargetArch(target: string): Architecture {
 }
 
 export function parseCompileFlags(cptVersion: cpt.Version, args: string[], lang?: string): CompileFlagInformation {
+  const require_standard_target = (cptVersion < cpt.Version.v5);
```

Comment:
I'm looking through the change history for cpptools-api and it looks like this should have been made optional in v4.  Cpptools requires v5 for standard to be optional though.

I think the bug is actually in cpptools `Client.isSourceFileConfigurationItem` requires v5 for intelliSenseMode and standard to be optional, but at this point we probably have to leave it alone.

Reply:
It's okay if a value sent for IntelliSenseMode for v4 even though it's optional. If the incorrect value is sent to cpptools, cpptools v1.3.0 can correct it from the #defines from compiler query.

From this PR https://github.com/microsoft/vscode-cpptools-api/pull/28, IntelliSenseMode and Standard were made optional when cpptools-api version was changed to 4.0.2, so I just required these properties for v4 or under.